cmd/jujud/agent/controllercharm.go

```diff
@@ -34,95 +35,170 @@ import (
 
 const controllerCharmURL = "ch:juju-controller"
 
-func (c *BootstrapCommand) deployControllerCharm(st *state.State, cons constraints.Value, charmPath string, channel charm.Channel, isCAAS bool, unitPassword string) (resultErr error) {
+type controllerCharmArgs struct {
+	cons         constraints.Value
+	charmPath    string
+	channel      charm.Channel
+	isCAAS       bool
+	unitPassword string
+	sshProxyPort int
+}
+
+type controllerCharmDeployer struct {
+	controllerCharmArgs
+
+	// The values below are populated during the deployment process.
+
+	// machine is only populated for IAAS deployments and contains
+	// info about the machine where the controller is deployed.
+	machine           *state.Machine
+	arch              string
+	base              corebase.Base
+	controllerAddress string
+	source            string
+	controllerUnit    *state.Unit
+	dataDir           string
+}
+
+func (c *BootstrapCommand) deployControllerCharm(st *state.State, args controllerCharmArgs) error {
```

Comment:
Move this out from between your `controllerCharmDeployer` declarations. Makes for junky reading.